### ISSUE TITLE
Visit object got deleted but events exist.

### ISSUE THREAD

**Issue by anandvc** [ankane/ahoy#100]
When I run a query like this:

```
 Ahoy::Event.includes(:visit).select{|x| x.visit.nil?}.count
```

I expect the result to say zero because every event should belong to a visit. However, I get a non-zero result:

```
=> 503
```

What's interesting is that:

```
Ahoy::Event.where(:visit_id=>nil).count
   (16.1ms)  SELECT COUNT(*) FROM `ahoy_events` WHERE `ahoy_events`.`visit_id` IS NULL
=> 0
```

Something's happening where the Visit object is getting deleted. I'm not sure why. I'm not touching it. I suspect it's a bug, but I don't know what helpful information I can provide to even fix it. Any insight will be appreciated.

**Comment by AdamQuadmon** [comment]
+1

**Comment by srfoster** [comment]
Seeing the same issue.  Any advice?

**Comment by anandvc** [comment]
I haven't found a solution yet.

**Comment by ankane** [comment]
There's nothing that should be deleting visit objects in the code, so it seems like they were probably never created in the first place. A bug was recently fixed in #189 which could have caused this.
